src/protocol/reading.rs

```diff
@@ -14,14 +20,32 @@ impl Reading for InnerNode {
         Default::default()
     }
 
+    /// Terminates WebSocet packets, decodes and forwards [AlgoMessage] payload to synthetic node's inbound queue.
     async fn process_message(&self, source: SocketAddr, message: Self::Message) -> io::Result<()> {
-        info!(parent: self.node().span(), "got a message from {}: {:?}", source, message);
+        let span = self.node().span();
+        debug!(parent: span, "got a message from {}: {:?}", source, message);
+
+        // Only binary messages are expected.
+        if message.opcode() != websocket_codec::Opcode::Binary {
+            warn!(parent: span, "not a binary opcode");
+            return Ok(());
+        }
+
+        // TODO(Rqnsom): maybe mv this algocodec to our inner node, encapsulated in a RefCell
+        let mut algomsgcodec = AlgoMsgCodec::new(span);
+        let mut bytes =
+            BytesMut::try_from(message.into_data().as_ref()).map_err(|_| ErrorKind::InvalidData)?;
+        let msg = algomsgcodec
+            .decode(&mut bytes)
+            .expect("couldn't decode the message")
+            .expect("message not found");
```

Comment:
we should never panic on read failures in `Reading::process_message`

Reply:
Aha right! Will fix!